app.js

```diff
@@ -19,16 +21,50 @@ mongoose.connect('mongodb://localhost:27017/mestodb', {
   useUnifiedTopology: true,
 });
 
+app.use(cookieParser());
+
+app.use(requestLogger);
+
 app.use(cookieParser());
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
+app.post('/signup', celebrate({
+  body: Joi.object().keys({
+    email: Joi.string().required().email(),
+    password: Joi.string().required().min(8),
+    name: Joi.string().required().min(2).max(30),
+    about: Joi.string().required().min(2),
+  }),
+}), createUser);
+
 app.post('/signin', login);
-app.post('/signup', createUser);
 
 app.use(auth);
 app.use('/', users);
 app.use('/', cards);
 
-app.listen(PORT, () => {});
+app.get('/crash-test', () => {
+  setTimeout(() => {
+    throw new Error('Что то пошло не так, загрузка сервера  прервана');
+  }, 0);
+});
+
+app.post('/signin', celebrate({
+  body: Joi.object().keys({
+    email: Joi.string().required().email(),
+    password: Joi.string().required().min(8),
+  }),
+}), login);
+
 app.all('/*', (req, res) => res.status(404).send({ message: 'Запрашиваемый ресурс не найден' }));
+
+app.use((err, req, res, next) => {
+  res.status(err.statusCode || 500).send({ message: err.message });
+  next();
+});
+
+app.use(errorLogger);
+app.use(errors());
```

Comment:
Нужно исправить: обработка роутов `celebrate` не срабатывает.
Это происходит из-за того, что [централизованный обработчик ошибок](https://github.com/MaximMaslovVik/15_sprint/blob/9aae875d58b5031fbc6c13ac1f5332065778c271/app.js#L62-L65) назначен **ранее** чем срабатывает [обработчик `celebrate`](https://github.com/MaximMaslovVik/15_sprint/blob/9aae875d58b5031fbc6c13ac1f5332065778c271/app.js#L68).

Чтобы это исправить, нужно централизованный обработчик опустить вниз:
 ```suggestion
app.use(errorLogger);
app.use(errors());
app.use((err, req, res, next) => {
  res.status(err.statusCode || 500).send({ message: err.message });
  next();
});
```